solr/solrj/src/java/org/apache/solr/client/solrj/impl/BaseHttpClusterStateProvider.java

```diff
@@ -248,15 +240,19 @@ > getCacheTimeout()) {
               + "succeeded in obtaining the cluster state from none of them."
               + "If you think your Solr cluster is up and is accessible,"
               + " you could try re-creating a new CloudSolrClient using working"
-              + " solrUrl(s) or zkHost(s).");
+              + " solrUrl(s).");
     } else {
       return this.liveNodes; // cached copy is fresh enough
     }
   }
 
+  private boolean updateLiveNodes(List<String> liveNodes) {
+    return updateLiveNodes(Set.copyOf(liveNodes));
+  }
+
   private boolean updateLiveNodes(Set<String> liveNodes) {
```

Comment:
why does updateLiveNodes take a Set as an argument; can't it simply be a Collection?  The Set is forcing you to do a conversion and that which changes the order, and the order may be deliberate.  It's not worth overloading the method for either.